src/global/Layouts/HamburgerMenu/HamburgerMenu.tsx

```diff
@@ -0,0 +1,179 @@
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+`;
+const HamburgerIcon = styled(Icon)`
+  margin-left: 28px;
+`;
+
+const TextInHamburger = styled.div`
+  margin-left: 16px;
+  color: ${props => props.theme.colors.utils.text.dark};
+  && {
+    text-decoration: none;
+  }
+  ${props => props.theme.typography.subtitle2};
+`;
+
+const LinkRouterStyle = styled(NavLink)`
+  display: flex;
+  align-items: center;
+  color: ${props => props.theme.colors.main.primary}
+  height: 50px;
+  width:100%;
+  border-radius: 4px;
+  text-decoration: none
+  :hover
+  {
+  background-color: rgba(85, 133, 255, 0.2);
+  }
+`;
+
+const DrawerWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  padding: 12px 16px;
+  font-size: 18px;
+  background-color: ${props => props.theme.colors.utils.background.mid.color};
+`;
+
+const MenuDivider = styled.div`
+  height: 24px;
+  background-color: ${props => props.theme.colors.utils.text.dark};
+`;
+
+const HamburgerMenuIcon = styled(Icon).attrs({
+  svgLink: HamburgerMenuSvg,
+})`
+  margin-right: 36px;
+  background-color: ${props => props.theme.colors.utils.background.light};
+  height: 20px;
+  width: 30px;
+`;
+
+const StyledDrawer = styled(Drawer).attrs({
+  width: 'calc(100% - 56px)',
+})`
+  .ant-drawer-body {
+    padding: 0;
+  }
+  &&& .ant-drawer-content-wrapper {
+    height: auto;
+  }
+`;
+
+const SearchPositionBox = styled.div`
+  position: relative;
+  display: flex;
+  align-items: center;
+  width: 100%;
+`;
+
+const SearchImg = styled.img`
+  position: absolute;
+  margin-left: 8px;
+`;
+const SearchInputStyled = styled.input`
+  width:100%
+  padding-left:32px;
+  background: #9db9ff;
+  height: 32px;
+  margin-right: 28px;
+  border: 1px solid rgba(255, 255, 255, 0.1);
+  box-sizing: border-box;
+  border-radius: 4px;
+  :: placeholder{
+  ${props => props.theme.typography.body2};
+  color: rgba(50, 50, 93, 0.5);
+  }
+  :focus  {outline: none;}
+  :hover {border:1px solid ${props => props.theme.colors.utils.background.light}}
+`;
+const RightProfile = styled.img`
+  width: auto;
+`;
+
+const LogoutButton = styled.div`
+  display: flex;
+  align-items: center;
+  color: ${props => props.theme.colors.main.primary}
+  height: 50px;
+  width:100%;
+  border-radius: 4px;
+  text-decoration: none
+  :hover
+  {
+  background-color: rgba(85, 133, 255, 0.2);
+  }`;
+
+export const HamburgerMenu = ({ isWithHamburger, isWithSearch }: HamburgerMenuType) => {
+  const themeContext = useContext(ThemeContext);
+  const [isMenuOpened, setIsMenuOpened] = useState(false);
+  const history = useHistory();
+  const [logout] = useMutation<SignOutResponse>(SIGN_OUT);
+
+  return (
+    <>
+      {isWithHamburger && (
+        <HamburgerMenuStyled>
+          <HamburgerMenuIcon color={themeContext.colors.main.tetiary} onClick={() => setIsMenuOpened(true)} />
+          <StyledDrawer placement="left" closable={false} onClose={() => setIsMenuOpened(false)} visible={isMenuOpened}>
```

Comment:
Ok, so the size of this component can be reduce by extracting StyledDrawer together with MenuDivider and DrawerWrapper. You can have component called Drawer so that it can be build like this:
`<StyledDrawer>
<HeaderDivider/>
<DrawerWrapper>
{children}
</DrawerWrapper>
</StyledDrawer>
`